test/Connection/Http1ConnectionTest.php

```diff
@@ -124,6 +125,66 @@ public function testUpgrade(): \Generator
         $this->assertSame([], (yield $response->getTrailers())->getHeaders());
     }
 
+    public function testTransferTimeout(): \Generator
+    {
+        $this->setMinimumRuntime(500);
+        $this->setTimeout(600);
+
+        [$server, $client] = Socket\createPair();
+
+        $connection = new Http1Connection($client);
+
+        $request = new Request('http://localhost');
+        $request->setTransferTimeout(500);
+
+        /** @var Stream $stream */
+        $stream = yield $connection->getStream($request);
+
+        $server->write("HTTP/1.1 200 Continue\r\nConnection: keep-alive\r\nContent-Length: 8\r\n\r\ntest");
+
+        /** @var Response $response */
+        $response = yield $stream->request($request, new NullCancellationToken);
+
+        $this->assertSame(200, $response->getStatus());
+
+        try {
+            yield $response->getBody()->buffer();
+            $this->fail("The request should have timed out");
+        } catch (TimeoutException $exception) {
+            $this->assertStringContainsString('transfer timeout', $exception->getMessage());
+        }
+    }
+
+    public function testInactivityTimeout(): \Generator
+    {
+        $this->setMinimumRuntime(500);
+        $this->setTimeout(600);
+
+        [$server, $client] = Socket\createPair();
+
+        $connection = new Http1Connection($client);
+
+        $request = new Request('http://localhost');
+        $request->setInactivityTimeout(500);
+
+        /** @var Stream $stream */
+        $stream = yield $connection->getStream($request);
+
+        $server->write("HTTP/1.1 200 Continue\r\nConnection: keep-alive\r\nContent-Length: 8\r\n\r\ntest");
```

Comment:
Should this test send two chunks to differentiate it from the transfer timeout test case?

Reply:
Good idea, updated.